packages/select/src/index.ts

```diff
@@ -64,32 +74,80 @@ const validateConfig = (config: SelectConfig): void => {
 
 const validateSelector = (selectorFactories, selectorName, model): void => {
 	if (process.env.NODE_ENV !== 'production') {
-		if (typeof selectorFactories![selectorName] !== 'function') {
+		if (typeof selectorFactories?.[selectorName] !== 'function') {
 			throw new Error(
 				`Selector (${model.name}/${selectorName}) must be a function`
 			)
 		}
 	}
 }
 
-const createSelectPlugin = (config: SelectConfig = {}): Plugin => {
+/**
+ * Todo: could we extend types of standard createModel?
```

Comment:
@tianzhich

Reply:
I was thinking if we create a modelCreator for each plugin for that purpose. How can we use them together?
Could we add a loose index type for `createModel` just like this:

```ts
export const createModel: <
	RM extends Models<RM>
>() => <
	R extends ModelReducers<S>,
	BR extends ReduxReducer<BS>,
	E extends ModelEffects | ModelEffectsCreator<RM>,
	S,
	BS = S
>(mo: {
	name?: string
	state: S
	reducers?: R
	baseReducer?: BR
	effects?: E
	[index: string]: any
}) => {
	name?: string
	state: S
	reducers: R
	baseReducer: BR
	effects: E
        [index: string]: any
} = () => (mo): any => {
	const { reducers = {}, effects = {} } = mo

	return {
		...mo,
		reducers,
		effects,
	}
}
```

So our users can add their own model configs if they needed when creating a new plugin. But cause the type is any, so they need to use `as` to get type completion.

Reply:
Or we can collect their config into a prop called `configs` so we can pass the type param to get the completion.

Reply:
like `createModel<RootModel, ExtraSelectorsModel<RootModel>>()({})`?

Reply:
I am not sure hot to extend native createModel for plugins. If we want the automation from selectors just like effects, seems the only way to work is yours. But if we have another plugin which also need model config. Are we create a helper `createModelWithXXX`? How could we use them together?

Or we can tell our plugin contributers to modify `createModel` in core and add these config definitions, if needed.